### PR TITLE
T14257

### PR DESCRIPTION
**Pull request by katelk28**
**Phabricator Issue:**
https://phabricator.endlessm.com/T14257

Yay my first PR! Wooooohoooo 
o<|:~)--/--<

### CODE REVIEW

File: data/css/modules/card/_media.scss [hunk 1]
```diff
@@ -1,35 +1,30 @@
...
 
-    separator {
-        background-color: $black-20-color;
-        min-height: 1px;
-    }
-
     &__caption {
-        font-size: .81em;  /* 13pt * 1/12 * 0.75 DPI correction factor */
-        font-family: $display-font;
+        font-size: 20px;
```
Comment (nedrichards) [inline data/css/modules/card/_media.scss]: Drive by review with no actual knowledge - but do we want to keep using em's rather than px here?

File: data/css/modules/layout/_sidebar.scss [hunk 2]
```diff
@@ -6,57 +6,49 @@
...
         .CardTitle {
             margin: 0px;
-            padding: 6px;
-            border: 2px solid transparentize(white, 1);
-            background-color: transparentize(white, 1);
-            transition: background-color 250ms ease-in-out,
-                        background-image 250ms ease-in-out,
-                        border 250ms ease-in-out;
+            padding: 10px;
+            // border-bottom: 1px solid transparentize(black, 1 - 0.7);
```
Comment (nedrichards) [inline data/css/modules/layout/_sidebar.scss]: Any reason why we need to keep commented out code?

File: data/css/modules/layout/_sidebar.scss [hunk 3]
```diff
@@ -6,57 +6,49 @@
...
+                background: transparentize(black, 1 - 0.5);
+                font-weight: 700;
 
                 .Card__title {
-                    color: $primary-dark-color;
-                    transition-property: none;
+                    color: white;
+                    transition-property: color;
+                    transition-timing-function: ease;
+                    transition-duration: 0.5s;
```
Comment (nedrichards) [inline data/css/modules/layout/_sidebar.scss]: Nice! Have we tested perf on the mini or another lower end device for this? Or will it be disabled by default on those devices, I forget how we performance bucket.

File: data/templates/css/wikihow.scss [hunk 4]
```diff
@@ -22,39 +21,32 @@ $xga: 1024px * 36 / 56;
...
+    font: 20px $font-primary;
 }
 
 body {
-    background: #ececec image-url("noise.png");
-}
-
-img {
-    vertical-align: middle;
+    background: #fff;
```
Comment (nedrichards) [inline data/templates/css/wikihow.scss]: Stupid consistency question, but in some files we go for #fff and in some files, white. Any particular reason?

File: data/templates/css/wikihow.scss [hunk 5]
```diff
@@ -221,16 +261,17 @@ h2 + ol, h3 + ol, .steps_list_2 {
...
 
-//
+
+//////////////////////////////
 // Responsive
-//
+//////////////////////////////
 
-@media (min-width: $vga) {
+@media (min-width: $vga) {  // FIXME: Haven't tested for vga yet
```
Comment (nedrichards) [inline data/templates/css/wikihow.scss]: Do we need to fix this before merging? Or is it OK.

File: data/templates/css/wikimedia.scss [hunk 6]
```diff
@@ -387,15 +458,15 @@ html {
...
 
-//
+//////////////////////////////
 // Responsive
-//
+//////////////////////////////
 
-@media (min-width: $vga) {
-    .mw-body {
+@media (min-width: $vga) {  // FIXME: Haven't tested for vga yet
```
Comment (nedrichards) [inline data/templates/css/wikimedia.scss]: Same as above, do we need to fix this before merging the PR?

File: data/templates/css/wikimedia.scss [hunk 7]
```diff
@@ -411,6 +415,7 @@ html {
     .mw-body {
         $in-cols: 100% / 36;
         padding: 30px $cols * 2;
+        // padding: 10px 10px 10px 10px;
```
Comment (tchx84) [inline data/templates/css/wikimedia.scss]: Hi! If these are not necessary why leaving them :)?

File: data/templates/css/wikimedia.scss [hunk 8]
```diff
@@ -283,28 +326,87 @@ html {
...
+    	
+    	// th {
+     //    	font: 700 1.5em $font-heading !important; /* 24px / 16px/em */
+     //    	background-color: $tan-color !important;
+     //        padding: 22px;
+     //    }
+
+     //    th a.eos-no-link {
+     //    	font: 700 1.5em $font-heading !important; /* 24px / 16px/em */
+     //    }
```
Comment (tchx84) [inline data/templates/css/wikimedia.scss]: If there are no needed, then just remove it.

File: data/templates/css/wikimedia.scss [hunk 9]
```diff
@@ -283,28 +326,87 @@ html {
...
+
         // &.geography
         *[style*="nowrap"] {
             white-space: normal !important;
         }
         td, th {
-            border-top: 1px solid $gray-light;
-            padding: 0.4em 0.6em;
+            // border-top: 1px solid $gray-light;
+            // padding: 0.4em 0.6em;
```
Comment (tchx84) [inline data/templates/css/wikimedia.scss]: same here!

File: data/templates/css/wikihow.scss [hunk 10]
```diff
@@ -8,10 +8,16 @@
...
-$font1-composite: "DejaVu Sans";
-$font2: "Montserrat", sans-serif;
+$font-primary: "Lato", sans-serif;
+$font-primary-composite: "Roboto";
+$font-secondary: "Fira Sans", sans-serif;
+
+
+
+
+
```
Comment (tchx84) [inline data/templates/css/wikihow.scss]: we probably don't need all these blank lines :)

File: data/templates/css/wikihow.scss [hunk 11]
```diff
@@ -145,25 +193,26 @@ $rhythm: 22px;
 h2 + ol, h3 + ol, .steps_list_2 {
     $in-cols: 100% / (36 - 2 * 2);
     > li {
-        border-left: 1px solid $gray-lighter;
+        // border-left: 1px solid $gray-lighter;
         clear: both;
-        margin-bottom: $rhythm;
-        padding-left: $in-cols * 2;
+        margin-bottom: 50px;;
+        // padding-left: $in-cols * 2;
```
Comment (tchx84) [inline data/templates/css/wikihow.scss]: same here.

File: data/templates/css/wikihow.scss [hunk 12]
```diff
@@ -145,25 +193,26 @@ $rhythm: 22px;
...
         @include box-sizing(border-box);
         @include pie-clearfix;
     }
     .mwimg {
         margin: 0 auto;
         margin-bottom: $rhythm;
         position: relative;
-        width: $in-cols * 30;
+        // width: $in-cols * 30;
```
Comment (tchx84) [inline data/templates/css/wikihow.scss]: same here, and for the others.

File: data/css/modules/card/_media.scss [hunk 13]
```diff
@@ -22,7 +22,6 @@
         font-size: .62em;  /* 10pt * 1/12 * 0.75 DPI correction factor */
         font-family: $display-font;
         color: $accent-light-color;
-        transition: color 250ms ease-in-out;
```
Comment (tchx84) [inline data/css/modules/card/_media.scss]: The commit description mentions changes in the sidebar and text styling, but the commit contains many changes that are not related to that. Including changes to the non-GTK styling sheets.

Can you double check whether these changes belong here? and if they do, elaborate a bit more in the description to explain what this commit achieves.

File: data/css/modules/layout/_sidebar.scss [hunk 14]
```diff
@@ -7,7 +7,8 @@
         .CardTitle {
             margin: 0px;
             padding: 10px;
-            border-bottom: 2px solid transparentize(black, 1 - 0.1);
+            border-bottom: 1px solid transparentize(black, 1 - 0.2);
+            transition: all 0.2s ease;
```
Comment (tchx84) [inline data/css/modules/layout/_sidebar.scss]: I think some changes don't corresponding to this commit? Please double check! :)

File: data/templates/css/wikihow.scss [hunk 15]
```diff
@@ -368,14 +364,31 @@ h2 + ol, h3 + ol, .steps_list_2 {
...
+        .eos-article-title {
+            padding: 20px 0px 40px 0px; // 40px because of top padding of 20px
         }
     }
-}
+
+    // .section li,
+    // .section ul {
+    //     padding: 20px;
+    // }
```
Comment (tchx84) [inline data/templates/css/wikihow.scss]: Same thing re. leaving these here.

File: data/css/modules/card/_media.scss (hunk 1 above)
Comment (tchx84) [inline data/css/modules/card/_media.scss]: @mattdangerw your opinion is more than welcome here :)

File: data/templates/css/wikihow.scss [hunk 16]
```diff
@@ -266,31 +261,17 @@ h2 + ol, h3 + ol, .steps_list_2 {
...
-    padding-top: 40px;
-    padding-bottom: 40px;
-    margin-top: 40px;
-    border-top: 1px solid $gray-lighter;
-    clear: both;
-    p {
-        width: 100% !important;
-    }
-}
-
```
Comment (tchx84) [inline data/templates/css/wikihow.scss]: I noticed you added this earlier, maybe it would better to aggregate these changes?

File: data/css/modules/layout/_sidebar.scss [hunk 17]
```diff
@@ -18,20 +18,14 @@
             }
 
             &:hover {
-                // background: transparentize(white, 1 - 0.05);
```
Comment (tchx84) [inline data/css/modules/layout/_sidebar.scss]: It would be better to aggregate these changes to the commit where you commented these line.

File: data/css/modules/card/_media.scss (hunk 1 above)
Comment (mattdangerw) [inline data/css/modules/card/_media.scss]: I think going `px` is fine here and lots of other places in a framework css dos the same.

The problem with `em` is since it cascades its a really bad fit for our modular setup. `2em` inside a module that set font size to be `1.5em` and your fonts are looking way different then in another modular hierarchy.

`rem` could be a possibility going forward, but its kinda useless unless we are doing some responsiveness with the the base font size which we are not. And that's out of scope for this issue anyway.

Comment (nedrichards) [inline data/css/modules/card/_media.scss]: lgtm

File: data/css/modules/card/_media.scss [hunk 18]
```diff
@@ -1,35 +1,30 @@
...
-    separator {
-        background-color: $black-20-color;
-        min-height: 1px;
-    }
-
     &__caption {
-        font-size: .81em;  /* 13pt * 1/12 * 0.75 DPI correction factor */
-        font-family: $display-font;
+        font-size: 20px;
+        font-family: Merriweather;
```
Comment (tchx84) [inline data/css/modules/card/_media.scss]: Just a reminder of what we talked on slack about using specific fonts in modules partials, we should ideally replace it with the proper scss font variables unless we really want all of our apps to use this specific font.

File: data/templates/css/wikimedia.scss [hunk 19]
```diff
@@ -43,6 +43,7 @@ $divider-color-composite: #959595;
 }
```
Comment (tchx84) [inline data/templates/css/wikimedia.scss]: Try to break your commit messages lines, you can find some good guidelines [here](https://github.com/endlessm/eos-documentation/wiki/Commit-Message-Practices)

File: data/templates/css/wikimedia.scss [hunk 20]
```diff
@@ -76,253 +77,317 @@ div[style*="float:right"] {
...
+            max-width: 100%;
+        }
+
+        h2, .firstHeading, h3, h4, h5 {
+        	color: $text-color-title;
+            line-height: 1.125;
+        }
+
+        h2, .firstHeading {
+        	font: 700 2.2rem $font-secondary;  // 44px // Root: 20px
```
Comment (tchx84) [inline data/templates/css/wikimedia.scss]: watch out with tabs :) we want spaces here! (and everywhere else)

File: data/templates/css/wikimedia.scss [hunk 21]
```diff
@@ -76,253 +77,317 @@ div[style*="float:right"] {
...
+                width: 100% !important;
+                display: inline-block;
+                margin-bottom: 20px;
+
+                div, .gallerytext p, .gallerytextwrapper {
+                    width: auto !important;
+                }
+
+                .gallerytext {
+                	font-family: $font-secondary; // 16 px
```
Comment (tchx84) [inline data/templates/css/wikimedia.scss]: idem! No tabs!

Note that you can make a commit that fixes this, and then use the same interactive rebase trick to unify these fixes in this same commit.

File: data/templates/css/wikimedia.scss [hunk 22]
```diff
@@ -76,253 +77,317 @@ div[style*="float:right"] {
...
+                }
+                &:last-child:after {
+                    display: none;
+                }
             }
         }
-        &[style*="text-align: center"] {
-            td, th {
+        .wikitable {
+        	font-size: 0.8rem;	// 16px // Root: 20px
```
Comment (tchx84) [inline data/templates/css/wikimedia.scss]: Idem.

File: data/templates/css/wikimedia.scss [hunk 23]
```diff
@@ -76,253 +77,317 @@ div[style*="float:right"] {
...
-            display: inline;
-            margin: 0;
-        }
-        dd, li {
-            &:after {
-                content: " · ";
-                font-weight: bold;
+
+        tbody {
+    		font-size: 0.8rem;	// 16px // Root: 20px
```
Comment (tchx84) [inline data/templates/css/wikimedia.scss]: There are probably more, so just make sure to search for \t

File: data/templates/css/wikimedia.scss [hunk 24]
```diff
@@ -76,253 +77,317 @@ div[style*="float:right"] {
...
-        th {
-            font-weight: 700;
-        }
-        .maptable {
-            td, th {
-                border-top: 0 none;
+            li {
+                display: inline;
+                &:before {
+                    // content: " · ";
```
Comment (tchx84) [inline data/templates/css/wikimedia.scss]: what this selector do?

File: data/templates/legacy-article.mst [hunk 25]
```diff
@@ -11,13 +11,12 @@
     {{/title}}
     <section id="inside-content" class="section">
         <div>{{{body-html}}}</div>
-        <hr>
```
Comment (tchx84) [inline data/templates/legacy-article.mst]: I noticed you removed this, but also that you moved the disclaimer out of this section, wouldn't that be a problem considering that the selectors [here](https://github.com/endlessm/eos-knowledge-lib/pull/1267/commits/94249c5614659360e65fe4e47e1996337759d42b#diff-db16cecbc2ecc49d958cd74c4b659d68R526) assume that this going inside that section?

Please let me know if I am missing something :)

File: data/css/modules/card/_media.scss [hunk 26]
```diff
@@ -3,7 +3,7 @@
 
     &__caption {
         font-size: 20px;
-        font-family: Merriweather;
+        font-family: $body-font;
```
Comment (tchx84) [inline data/css/modules/card/_media.scss]: You can use interactive rebase to fix your previous commit (basically by moving this commit below the original commit and set it as fixup)

File: data/templates/css/wikimedia.scss [hunk 27]
```diff
@@ -431,15 +496,48 @@ html {
...
+    .wikitable,
+    tbody,
+    .thumbcaption,
+    .disclaimer {
+        font-size: 1.0rem !important;  // 20px // Root: 20px
     }
 }
+
+
+
```
Comment (tchx84) [inline data/templates/css/wikimedia.scss]: I didn't see this on wikihow articles though, so maybe is something related only to the wikipedia scss?

File: data/templates/legacy-article.mst (hunk 25 above)
Comment (katelk28) [inline data/templates/legacy-article.mst]: Good point, but as far as I know, `<hr>` and borders are different. 
In the .disclaimer class, I added a top-border, which adds a border to the .disclaimer div --- it's not manipulating the `<hr>`. They just look very similar.

File: data/templates/css/wikimedia.scss (hunk 24 above)
Comment (katelk28) [inline data/templates/css/wikimedia.scss]: My mistake. Forgot to take the comment off.

File: data/templates/css/wikimedia.scss (hunk 27 above)
Comment (katelk28) [inline data/templates/css/wikimedia.scss]: So this is actually a separate problem, and I've noticed that too.

Here's what happened: I made a design decision that, in composite mode, adds a `border-bottom` to the `h2` and `eos-article-title` elements, b/c it's very hard to differentiate between sections when the screen is that small.

BUT for some reason, maybe in the ingestor, when a section is hidden (like the "References" section or something), it seems that the `h2` element is still there in the HTML, even though it doesn't display anything. That's why the `border-bottom` is still applied.

I'm not sure if that's something that can be fixed with CSS, but I guess for now, since it's making that ugly bug, I will take out the `border-bottom` until I come up with a better solution.

File: data/templates/legacy-article.mst (hunk 25 above)
Comment (tchx84) [inline data/templates/legacy-article.mst]: I meant something else. What I meant is that before this patch we have:
```
<inside-content>
  <disclaimer>
</inside-content>
```

After this patch we have:

```
<inside-content>
  ...
</inside-content>
<disclaimer>
```

But, the CSS selectors still assume the previous structure [here](https://github.com/endlessm/eos-knowledge-lib/pull/1267/commits/cac136f4c92bc190d9eb830154bb58aa0f23d377#diff-db16cecbc2ecc49d958cd74c4b659d68R521), and I this is correct that selector should not be working.

**Comment by tchx84** [comment]
Looking better now, just noticed one more thing though:

screenshot from 2016-12-13 13-15-45

As you can see in the screenshot, there is a big gap between title and the content.

I tested with random articles from encyclopedia, astronomy and howto apps, but only noticed this in astronomy, some examples are "earth science" and "physical cosmology" articles. Try comparing with master (let me know if you need help getting these shards).

Also, since not all the articles use the exact same structure is not always possible to cover every single case. Sometimes we fix something and possibly break other things, so I would suggest to give a quick try and see if you add a selector for cover this case (unless you find something evidently wrong).

**Comment by tchx84** [comment]
**CR PASS**
